Objetivo 2 [IV-22-23]

LuisMart7 opened the pull request:
# Estructura de repositorio

* [ ] ¿He seguido las mejores prácticas en nombre de las clases y ficheros y disposición de los mismos?

# Análisis de problema

* [ ] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo que se ha creado?
* [ ] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor, una entidad o un agregado?
* [ ] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo objeto valor que no dependa de ningún otro (y que sea la base de muchos otros)?

# Planificación y Programación

* [ ] ¿Los issues representan un problema, y no una tarea?
* [ ] ¿Todos los mensajes de commit explican el cambio, y no se limitan a repetir el nombre del fichero que se ha cambiado?
* [ ] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [ ] ¿Se ha hecho una revisión real del código para comprobar que todos los atributos y funciones creadas están respaldadas por una HU?
* [ ] ¿Todos los issues creados están asignados a una HU?
* [ ] ¿Todos los cambios en el código están asignados a un issue al que se referencia en un commit? Los issues sobre los que estoy trabajando, ¿han sido asignados al primer milestone?
* [ ] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [ ] ¿Son los milestones sobre los que estoy trabajando productos mínimamente viables? ¿O tengo que solicitar al product manager que cree milestones adicionales o precise de qué producto se trata?
* [ ] ¿Se ha comprobado que el código entregado sea sintácticamente correcto?

LuisMart7 committed:
Create README.md

danielsp13 commented:
Ánimo con ello. Yo estoy por aquí revisando lo que hagas 👍🏼

LuisMart7 committed:
Avanza en la tarea #11 añadiendo pregunta.py

Se ha creado el fichero donde se implementa la clase Pregunta con sus distintos campos [#11](https://github.com/danielsp13/SuperCatch/issues/11)

LuisMart7 committed:
Avanza en la tarea #11 añadiendo iv.yaml

Se crea el fichero donde se plasma el lenguaje y las nuevas entidades que se conforman en [#11](https://github.com/danielsp13/SuperCatch/issues/11)

LuisMart7 commented:
Cuando puedas le echas un ojo a los issues @danielsp13 :)

danielsp13 commented:
Buenos días Luis, estoy revisando

danielsp13 commented:
Por el momento, eso es todo, estaré atento a más cambios que hagas y de forma más activa para que puedas trabajar de inmediato, ya que el fin de semana no he podido.

LuisMart7 committed:
Create alumno.py

Avanza en la tarea [danielsp13#11](https://github.com/danielsp13/SuperCatch/issues/11) creando el archivo alumno.py

LuisMart7 committed:
Avanza en la tarea [danielsp#11](https://github.com/danielsp13/SuperCatch/issues/11) añadiendo examen.py

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/11 completando pregunta.py

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/11 definiendo la clase Pregunta

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/11 modificando README.md

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/11 añadiendo respuesta.py

JJ commented:
Mirad las mejores prácticas en mensajes de commit. Estos no deben repetir lo obvio.

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/11 definiendo clase Respuesta

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/11 añadiendo entidades Pregunta y Respuesta

JJ reviewed (commented):
Por favor, mirad los comentarios por ambas partes.

LuisMart7 commented on src/respuesta.py:
```diff
@@ -0,0 +1,21 @@
+@dataclass
+class Respuesta:
+
+    numero: int
```
De acuerdo. Dejaré el issue abierto para definir de manera correcta las clases Pregunta y Respuesta relacionadas con la facilidad de corrección por parte del profesor y abriré otro para llevar a cabo los PMVs que satisfagan la HU03.

JJ replied:
Los issues no definen nada, plantean problemas. Por favor, trata de verlo y enfocarlo de esa forma, porque será más sencillo alcanzar el resultado.

danielsp13 commented:
Asigné este PR al Milestone [M0](https://github.com/danielsp13/SuperCatch/milestone/1) tal y como se requiere para que superes este objetivo @LuisMart7

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/14 nombrando las clases necesarias

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/15 completando la clase Alumno

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/15 completando la clase Alumno

LuisMart7 committed:
Avanza en la tarea https://github.com/danielsp13/SuperCatch/issues/14 definiendo la clase ResolucionExamen

LuisMart7 committed:
Modifica README.md definiendo todas las clases con sus distintos campos

LuisMart7 commented:
@danielsp13 listo para revisar.

danielsp13 commented:
@LuisMart7 los commits que estás realizando no siguen el estándar que deberían.

En el mensaje de commit, debes indicar qué es lo que has hecho "ej: implementación de getters y setters en fichero.py", y además añadir una descripción de por qué y/o para qué.

En cuanto al código. Te lo revisaré en cuanto pueda.

danielsp13 commented:
Luis, mañana a primera hora empiezo con la revisión. Hoy no me va a ser posible.

danielsp13 commented on src/pregunta.py:
```diff
@@ -0,0 +1,33 @@
+    def get_posicion_palabra(self, palabra: str):
+        return self.palabras_clave.index(palabra)
+
+    def aniadir_palabra_clave(self, palabra: str):
+        self.palabras_clave.append(palabra)
```
No controlas que la palabra que se introduce realmente pertenece a la respuesta correcta. Imagínate que alguien es capaz de modificar este código, podría añadir palabras que podrían hacer que el algoritmo valorase de forma más positiva la respuesta otorgada por el alumno.

danielsp13 commented on src/resolucionExamen.py:
```diff
@@ -0,0 +1,28 @@
+    preguntas: list
+    respuestas: list
+
+    def __init__(self):
+        self.id_examen = 0
+        self.preguntas = []
+        self.respuestas = []
```
Este constructor no lo considero necesario.

LuisMart7 replied:
Corregido @danielsp13

danielsp13 commented on src/resolucionExamen.py:
```diff
@@ -0,0 +1,28 @@
+from respuesta import *
+from alumno import *
+
+class resolucionExamen:
+
+    id_examen: int
+    preguntas: list
+    respuestas: list
```
Además. ¿Cómo sé qué alumno ha respodido qué pregunta y cuál es su respuesta? Viendo el código y los atributos no me queda suficientemente claro. ¿Has realizado tests sobre esta clase?

JJ replied:
Solo comentar que este milestone no requiere test. Por favor, limitaros a lo que se busca en este milestone (y en este objetivo)

JJ replied:
> Me gustaría que añadieses aquí un atributo que identifique de qué examen estamos hablando (ej: nombre = "Tema 4").
> 

Si necesitas ese atributo tienes que identificarlo claramente en la historia de usuario sobre la que se esté trabajando.

> Además, te comento lo mismo que en el fichero de _alumno_, prefiero que estos atributos estén privados.

Esto son temas de implementación, que de la misma forma o están claramente identificados, en este caso en el milestone, o no le puedes pedir. En este objetivo se trata de que la persona propietaria del repo aprenda a escribir bien el material sobre el que va a trabajar la persona que lo hace.

danielsp13 reviewed (changes requested):
La valoración que te doy hasta el momento es la siguiente
- Documentación del análisis realizado sobre el problema. Planteamiento de issues y asociación con HU: :heavy_check_mark: 
- PMV correspondiente con los objetivos del milestone M0. :warning: . 
- Trabajo en la rama. :warning: 

En general considero que se aproxima bastante a lo que se requiere para que superes este objetivo pero necesito que hagas una serie de cambios.
1. Debes seguir las normas de commit. Cambia el mensaje que hay en ellos atendiendo a las buenas prácticas, porque si no, no te puedo aprobar los cambios. En cada commit debes indicar exactamente qué es lo que has hecho (ejemplo: "avanzo en tarea x, defino metodos de insercion en clase y.py") además de indicar una descripción de por qué y para qué.
2. Asegúrate de probar todo lo que implementes. El código debe ser fiable y no se puede permitir que se introduzcan ciertos erróneos (como el caso de añadir palabras clave en una respuesta correcta, sin verificar que dichas palabras pertenecen a ese texto), además del acceso directo a la modificación de atributos.
3. La clase resolución examen necesita todavía más estructura.
- Asegúrate de que se pueda identificar cada resolución, con el alumno que lo ha realizado.
- Prueba que realmente se verifican las historias de usuario.

LuisMart7 committed:
Feat(alumno.py): Add frozen dataclass and remove getters

Según lo hablado en https://github.com/danielsp13/SuperCatch/pull/10#discussion_r1027843960 y https://github.com/danielsp13/SuperCatch/pull/10#discussion_r1032136841 hemos cambiado la naturaleza de la clase a Dataclass con frozen (inmutable) y hemos eliminado las funciones get

LuisMart7 committed:
Feat(pregunta.py): Add frozen option and remove getters

Según lo hablado en https://github.com/danielsp13/SuperCatch/pull/10#discussion_r1027843960 y https://github.com/danielsp13/SuperCatch/pull/10#discussion_r1032140827 hemos cambiado la naturaleza de la Dataclass a frozen pues será inmutable y hemos eliminado los métodos get innecesasarios.

LuisMart7 committed:
Fix path of entities

Corrección de las rutas de la entidad resolucionExamen según lo mencionado en https://github.com/danielsp13/SuperCatch/pull/10#discussion_r1032134689

LuisMart7 commented on src/alumno.py:
```diff
@@ -1 +1,26 @@
+    dni: int
+    nombre: str
+    apellidos: str
+    calificaciones: list
```
Corregido @danielsp13

LuisMart7 commented on src/alumno.py:
```diff
@@ -1 +1,26 @@
+        return self.nombre
+    
+    def get_apellidos(self):
+        return self.apellidos
```
Corregido @danielsp13

danielsp13 commented:
Luis, analizando un poco todo lo que llevamos... quizá sería mejor abordar la representación del modelo desde una perspectiva diferente. Tanto para ti como para mi y para el profesor... 

Lo primero, tal y como me ha comentado el profesor en mi objetivo y en otras correcciones que ha hecho aquí, hay determinados issues que no plantean problemas , por ejemplo el último abierto...

Y con respecto a lo anterior, si has hecho código que refieren a esos issues, no es código que pueda definirse como válido...

Así que te recomiendo que le des una vuelta. Creo que ya tienes un buen análisis del dominio del problema de forma general, pero quizá habría que descomponerlo algo mejor atendiendo a una de las dos historias de usuario que tengo referenciadas en este milestone (la que tú elijas frente a la otra, es criterio tuyo). Una vez ahí, identifica las cuestiones simples que se necesitan modelar y a continuación prográmalo.

LuisMart7 committed:
Feat(resolucionExamen.py): Add dataclass definition and type field

Según lo hablado en https://github.com/danielsp13/SuperCatch/issues/14, hemos definido la clase resolucionExamen como un tipo dataclass además de añadir un campo a la misma que identifica si el examen es considerado una prueba realizada por el alumno o un examen real propuesto por el profesor.

LuisMart7 committed:
Feat(alumno.py): Add field Alumno and mark and remove getters

Según lo hablado en https://github.com/danielsp13/SuperCatch/issues/13, hemos añadido dos campos que definen la calificacion de la respuesta en comparación con la correcta y el alumno que realiza la misma. Hemos eliminado los métodos get pues no se pedían.

LuisMart7 committed:
Fix language name

Se corrige el nombre del lenguaje de python3 a python únicamente.

danielsp13 commented on src/resolucionExamen.py:
```diff
@@ -1,28 +1,35 @@
 
-    def aniadir_respuesta(self, resp:Respuesta):
+    def aniadir_respuesta(self, resp: Respuesta):
         self.respuestas.append(resp)
```
Puesto que esto no se menciona explícitamente en ninguna HU sobre las que hemos estado debatiendo, es mejor que estas funciones las elimines, y en el constructor coloques como argumentos la lista de preguntas y de respuestas...

LuisMart7 replied:
Perfecto.

danielsp13 commented on src/resolucionExamen.py:
```diff
@@ -1,28 +1,35 @@
+        self.tema = tema_examen
+        self.tipo = type
         self.preguntas = []
         self.respuestas = []
```
Sobre los constructores. El atributo de *tema_examen*, tampoco es algo que se mencione explícitamente en las historias de usuario, así que sería mejor que lo eliminases también.

LuisMart7 replied:
Recibido.

LuisMart7 replied:
Yo lo puse porque lo mencionamos a la hora de identificar el examen. También pienso que la corrección de exámenes se agilizaría teniendo en cuenta el tema del que se trata.

LuisMart7 committed:
Fix deleting inecessary set methods and name field

Según la https://github.com/danielsp13/SuperCatch/blob/main/docs/user-stories.md#black_nib-hu01-dolores-quiere-agilizar-la-tarea-de-correcci%C3%B3n-de-ex%C3%A1menes y https://github.com/danielsp13/SuperCatch/blob/main/docs/user-stories.md#black_nib-hu03-irene-necesita-entrenar-para-sus-ex%C3%A1menes, no es necesario crear métodos que añadan preguntas y respuestas además de la descripción del examen que se realiza pues no se tendrá constancia de un registro de exámenes.

LuisMart7 committed:
Refactor definition of classes

Se ha definido nuevamente la identidad de los productos mínimamente viables explicando su naturaleza de acuerdo a las soluciones propuestas por el desarrollador.

LuisMart7 committed:
Remove inecessary text

Se ha eliminado texto que estaba previamente.

danielsp13 commented on src/README.md:
```diff
@@ -7,3 +7,17 @@ Clase Respuesta: Al igual que anteriormente, hemos definido una clase inmutable
+## PREGUNTA:
+Se define un **objeto valor** que define la respuesta correcta asociada a una pregunta del examen propuesta. Para agilizar la corrección de acuerdo con la [HU01](https://github.com/danielsp13/SuperCatch/blob/main/docs/user-stories.md#black_nib-hu01-dolores-quiere-agilizar-la-tarea-de-correcci%C3%B3n-de-ex%C3%A1menes) hemos definido tres campos: *número* de la pregunta, *texto* asociado a la respu ...
+
+## RESPUESTA:
+Se define como una **entidad** que define la respuesta propuesta por el alumno a una pregunta del examen. Teniendo en cuenta que se ha de facilitar la corrección, se propone la definición de varios campos necesarios como: *número* de la pregunta respondida, *texto* escrito por el alumno, *calificacion* que se obtiene de esa respuesta y *alumno* que responde esa pregunta.
```
si `respuesta` es una *entidad*, en la clave del fichero `iv.yaml` debe estar esa respectiva entidad.

LuisMart7 replied:
Correcto. Gracias.

danielsp13 commented on src/README.md:
```diff
@@ -7,3 +7,17 @@ Clase Respuesta: Al igual que anteriormente, hemos definido una clase inmutable
 
 Clase Alumno: Esta entidad estará destinada a recoger los datos personales del alumno, además de las calificaciones que vaya obteniendo en los exámenes del curso.
+
+De acuerdo a las historias de usuario [HUO1](https://github.com/danielsp13/SuperCatch/blob/main/docs/user-stories.md#black_nib-hu01-dolores-quiere-agilizar-la-tarea-de-correcci%C3%B3n-de-ex%C3%A1menes) y [HU03](https://github.com/danielsp13/SuperCatch/blob/main/docs/user-stories.md#black_nib-hu03-irene-necesita-entrenar-para-sus-ex%C3%A1menes), se han definido cuatro PMVs que definen las peticion ...
```
4 pmv ... :thinking:

LuisMart7 replied:
Lo cambio para que se entienda mejor.

danielsp13 reviewed (changes requested):
@LuisMart7 , para que pueda aprobar este PR, si ya has considerado que tu solución efectivamente ha resuelto los problemas reflejados en los *issues*, es necesario cerrarlos con un commit de clausura:

~~~
closes #numIssue, ...
~~~
Toda tarea siempre se cierra con un commit. Necesito que lo hagas

LuisMart7 committed:
Add respuesta entity

Se añade la clase Respuesta con naturaleza de entidad.

LuisMart7 committed:
Fix error in the definition

Se ha modificado un error de concepto en la definición de las clases.

LuisMart7 committed:
Closes https://github.com/danielsp13/SuperCatch/issues/12, closes https://github.com/danielsp13/SuperCatch/issues/13, closes https://github.com/danielsp13/SuperCatch/issues/14, closes https://github.com/danielsp13/SuperCatch/issues/15, closes https://github.com/danielsp13/SuperCatch/issues/16

Cerramos la totalidad de issues pues se han solucionado todos.

LuisMart7 commented:
@danielsp13 listo para revisar :)

danielsp13 reviewed (approved):
Ahora, por mi parte está todo aprobado. Has trabajado muy bien @LuisMart7 , ha sido todo un placer colaborar contigo.

LuisMart7 committed:
Fix init methods

Se ha arreglado la asignación de campos en la inicialización debido a que se usa instancias del tipo Frozen.

LuisMart7 committed:
Fix init methods

Se ha arreglado la asignación de campos en la inicialización debido al uso de instancias del tipo Frozen.

LuisMart7 commented:
@JJ listo para revisar

danielsp13 commented:
@LuisMart7 , te he dicho varias veces que debes entregar el objetivo. Aunque el profesor no esté disponible estos días, por favor entrégalo y crea un PR en el repo del profesor.

LuisMart7 commented:
@JJ listo para revisar, disculpa Dani.

JJ commented:
> @JJ listo para revisar, disculpa Dani.

Cuando lo "envíes" como el resto de los objetivos, en tu línea de la tabla, como te ha dicho @danielsp13

LuisMart7 commented:
Ya @JJ listo para revisar. He creado el PR.

JJ commented:
La mayor parte de los commits no tienen referencia a ningún issue. Es imposible evaluar si lo que se ha incluido es correcto o no.

JJ reviewed (changes requested):
Por favor, mirad los errores frecuentes en este objetivo y los objetivos de aprendizaje de prácticamente todas las semanas anteriores, donde se hace énfasis en que se tiene que comenzar por la historia de usuario, porque si no se está avanzando en la solución de un problema.

LuisMart7 committed:
Feat(iv.yaml): Delete unnecesary line break

LuisMart7 commented on iv.yaml:
```diff
@@ -0,0 +1,5 @@
+lenguaje: python
+
+entidad:
+  - src/respuesta.py
```
Entiendo que es por el salto de línea porque el lenguaje y la lista de entidades sigue el formato.

JJ replied:
No es por el salto de línea. Los ficheros de un módulo en Python no se ponen en `src`.

LuisMart7 commented on src/pregunta.py:
```diff
@@ -0,0 +1,25 @@
+class Pregunta:
+
+    '''Clase que representa la respuesta correcta a una pregunta'''
+    numero: int
```
Numero de la pregunta que se responde en el examen pudiendo haber más de una pregunta. La respuesta significa el texto que responde a la perfección a la pregunta respondida. Sólo hay una respuesta correcta y es la propuesta por el profesor para tener un 10 en esa pregunta.

JJ replied:
¿no tendría que haberse planteado esto que dices en un issue?

LuisMart7 replied:
Lo definimos en el primer issue que realizamos que era demasiado genérico https://github.com/danielsp13/SuperCatch/issues/11

JJ replied:
No, no lo hiciste porque ese issue *no formula un problema*, como os he indicado repetidamente.

LuisMart7 replied:
Le abriré nuevamente un issue para determinar los mejores campos para la clase Pregunta. Igualmente, en otros issues se planteó el problema enfocado desde la clase Respuesta por eso si estaba hablado. En este caso, la forma que planteamos de resolverlo era una única respuesta correcta propuesta por el profesor y las palabras clave recogidas en esa respuesta con el fin de que fuera más fácil tener en cuenta si el alumno había insertado esas palabras para dar maoyr velocidad a la corrección.

LuisMart7 replied:
Definimos aquí los mejores campos para ambas clases para su correcta relación mutua https://github.com/danielsp13/SuperCatch/issues/12.

LuisMart7 replied:
He planteado el issue pero no sé si es necesario por lo tanto mantener la clase o en cambio añadir algún identificador a la respuesta que permita conocer el alumno que la realiza

JJ replied:
> Le abriré nuevamente un issue para determinar los mejores campos para la clase Pregunta. Igualmente, en otros issues se planteó el problema enfocado desde la clase Respuesta por eso si estaba hablado. En este caso, la forma que planteamos de resolverlo era una única respuesta correcta propuesta por el profesor y las palabras clave recogidas en esa respuesta con el fin de que fuera más fácil tener en cuenta si el alumno había insertado esas palabras para dar maoyr velocidad a la corrección.

Creo que no estás entendiendo correctamente qué se espera de ti en este objetivo. Tienes que hacer dos cosas:
1. Plantear (en issues) los problemas que representan las historias de usuario.
2. Escribir el código que resuelva esos problemas, siendo el código escrito un modelo del problema, según dice el milestone en el que estás trabajando.

No se trata de "los mejores campos" de "las clases". Se trata de un modelo del problema, que tienes que plasmar aquí y que @danielsp13 tiene que decidir si efectivamente aporta valor al cliente y contribuye a la solución de su problema. No se trata de *mantener la clase*. Se trata de plantear problemas. Si no has planteado problemas correctamente a partir de las historias de usuario, *todo el desarrollo es incorrecto*. Por favor, consulta el material impartido en clase, los errores frecuentes, y sobre todo el objetivo de aprendizaje repetido a lo largo de todas las semanas.

LuisMart7 commented on src/alumno.py:
```diff
@@ -0,0 +1,19 @@
+class Alumno:
+
+    '''Clase que representa un alumno que realiza el examen'''
+    dni: int
```
Se abre un issue por si hace falta esta clase.

LuisMart7 commented:
> La mayor parte de los commits no tienen referencia a ningún issue. Es imposible evaluar si lo que se ha incluido es correcto o no.

@JJ usted me menciono que en la cabecera del commit no era correcto poner el enlace del issue por eso añadí que issue tenía relacionado en la descripción del commit.

JJ commented:
> > La mayor parte de los commits no tienen referencia a ningún issue. Es imposible evaluar si lo que se ha incluido es correcto o no.
> 
> @JJ usted me menciono que en la cabecera del commit no era correcto poner el enlace del issue por eso añadí que issue tenía relacionado en la descripción del commit.

Dije justamente lo contrario: que tiene que estar en la cabecera, para que sea más fácil navegar desde esta página. Pero es que algunos commits no lo tienen ni en la cabecera ni en ningún otro lugar.

LuisMart7 commented:
> > > La mayor parte de los commits no tienen referencia a ningún issue. Es imposible evaluar si lo que se ha incluido es correcto o no.
> > 
> > 
> > @JJ usted me menciono que en la cabecera del commit no era correcto poner el enlace del issue por eso añadí que issue tenía relacionado en la descripción del commit.
> 
> Dije justamente lo contrario: que tiene que estar en la cabecera, para que sea más fácil navegar desde esta página. Pero es que algunos commits no lo tienen ni en la cabecera ni en ningún otro lugar.

Entonces lo entendí yo mal porque en el hackatón lo añadí y creía que me dijo que no era correcto. Lo tengo en cuenta para los posteriores.

JJ commented:
> > > > La mayor parte de los commits no tienen referencia a ningún issue. Es imposible evaluar si lo que se ha incluido es correcto o no.
> > > 
> > > 
> > > @JJ usted me menciono que en la cabecera del commit no era correcto poner el enlace del issue por eso añadí que issue tenía relacionado en la descripción del commit.
> > 
> > 
> > Dije justamente lo contrario: que tiene que estar en la cabecera, para que sea más fácil navegar desde esta página. Pero es que algunos commits no lo tienen ni en la cabecera ni en ningún otro lugar.
> 
> Entonces lo entendí yo mal porque en el hackatón lo añadí y creía que me dijo que no era correcto. Lo tengo en cuenta para los posteriores.

En ingeniería de software nada es correcto o incorrecto, sólo hay mejores o menos mejores opciones. Te he explicado por qué y en qué contexto (el de un PR) esta es una opción que facilita el trabajo de todo el mundo. Se trata de que entendáis lo que hacéis y por qué lo hacéis, no que recordéis lo que se dijo y de qué forma en esta u otra ocasión.

LuisMart7 committed:
Fix: Delete functionality methods https://github.com/danielsp13/SuperCatch/issues/12

Hemos eliminado las funcionalidades pues no se piden en ninguna historia de usuario asignada a este M0.

LuisMart7 commented:
@danielsp13 ayúdame con el nuevo issue que te he planteado por favor.

JJ commented:
> @danielsp13 ayúdame con el nuevo issue que te he planteado por favor.

Luis, no puedes plantear issues, sino soluciones. @danielsp13 te dirá si ayudan a la historia de usuario o no, en la revisión. En el desarrollo ágil no puede haber estas interacciones, porque si las hay se está empleando tiempo de dos personas, mientras que sólo una debería encargarse del tema. Recuerda además que este es *tu* objetivo, no el de Daniel.

LuisMart7 commented:
> > @danielsp13 ayúdame con el nuevo issue que te he planteado por favor.
> 
> Luis, no puedes plantear issues, sino soluciones. @danielsp13 te dirá si ayudan a la historia de usuario o no, en la revisión. En el desarrollo ágil no puede haber estas interacciones, porque si las hay se está empleando tiempo de dos personas, mientras que sólo una debería encargarse del tema. Recuerda además que este es _tu_ objetivo, no el de Daniel.

He planteado un problema en el issue dándole mi mejor solución para ello. Sólo quería que lo viera para que me diera el okay si lo veía bien.

JJ commented:
> > > @danielsp13 ayúdame con el nuevo issue que te he planteado por favor.
> > 
> > 
> > Luis, no puedes plantear issues, sino soluciones. @danielsp13 te dirá si ayudan a la historia de usuario o no, en la revisión. En el desarrollo ágil no puede haber estas interacciones, porque si las hay se está empleando tiempo de dos personas, mientras que sólo una debería encargarse del tema. Recuerda además que este es _tu_ objetivo, no el de Daniel.
> 
> He planteado un problema en el issue dándole mi mejor solución para ello. Sólo quería que lo viera para que me diera el okay si lo veía bien.

Ya hemos dicho repetidamente en clase que el OK es la aprobación al PR de la solución que tú hayas planteado. No es un examen que haya que aprobar, es un programa.

LuisMart7 committed:
Docs (alumno.py): Delete file

Hemos eliminado el archivo que define un alumno debido a que no se pide en la implementación inicial.

LuisMart7 committed:
Fix: Delete definition of module Alumno https://github.com/danielsp13/SuperCatch/issues/21

Hemos eliminado el módulo Alumno por lo que no tenemos que describirlo.

LuisMart7 committed:
Feat (respuesta.py): Remove Alumno and methods https://github.com/danielsp13/SuperCatch/issues/21

Hemos eliminado el campo Alumno además de los métodos innecesarios que no se piden.

LuisMart7 committed:
Closes https://github.com/danielsp13/SuperCatch/issues/21

Se cierra el issue debido a que hemos entendido que las historias de usuario no piden un módulo Alumno.

LuisMart7 committed:
Fix (README.md): Better definition

Hemos definido mejor el módulo Respuesta

LuisMart7 committed:
Docs (README.md): Change name and move to documentation

LuisMart7 commented:
@JJ listo para revisar.

danielsp13 reviewed (approved):
He visto todos los cambios que has realizado luis. La solución que has planteado en consecuencia  con los issues, me parece de momento suficiente para que pueda continuar con el siguiente hito.

JJ reviewed (changes requested):
No se han llevado a cabo prácticamente ninguno de los cambios pedidos ni por mi ni por @danielsp13. Recuerdo este comentario, uno de los que no se ha hecho.

danielsp13 commented:
Buenos días @LuisMart7 , es necesario terminar con este objetivo. Y para ello te voy a  dar una serie de instrucciones (que no difieren demasiado tampoco o prácticamente nada, con lo que haya indicado @JJ en la documentación de su repositorio, en la corrección de esta rama, y en otros proyectos).

Lo primero, reconozco que haber aprobado unos cambios que realmente, no me benefician a mi, ni tampoco a ti porque no superarás este objetivo, ha sido un fallo, además de tener dos historias de usuario asignadas en su momento, te ha perjudicado, así que mis disculpas.

Hay determinadas cosas que se explican en los errores frecuentes de todas las semanas, que son esencialmente útiles para la superación de este objetivo (y de todos):
* *"Deja que la historia de usuario sea tu guía"*: ¿Cuál es la HU que verdaderamente es importante para desarrollar un PMV? Ahora que ya está especificado correctamente en el milestone de todos modos, todos los problemas deberían ser extraídos de #5 .
* *"El desarrollo del Objetivo2, debe verse enfocado también al desarrollo del Objetivo4"*: Quizá esto es lo más clave de todo. ¿Realmente en este punto necesito entidades? ¿Necesito preguntas, alumnos, exámenes? Es MUY recomendable también leer las tareas que deben verificar los PR porque si te fijas, hay un item con este enunciado en este PR: **"¿He propuesto un producto mínimamente viable, que en muchos casos será un solo objeto valor que no dependa de ningún otro (y que sea la base de muchos otros)?"**.
* *Simpleza*: En vez de ofuscarse desarrollando código con clases y módulos, etc, en relación con el anterior punto, a veces sólo hay que centrarse en lo más simple para poder continuar con el desarrollo en fases posteriores.

danielsp13 commented:
Así que en relación con esto, @LuisMart7 , requiero que empieces desde el principio, yo voy a proceder a marcar todos los *issues* planteados hasta ahora como inválidos, porque primero creo que entorpecen más que ayudar, y segundo, porque no son correctos. 

La resolución de este objetivo en particular para este proyecto, no es difícil, en serio. Te recomiendo también que observes lo que te he comentado anteriormente, y todos los errores frecuentes relacionados con este obetivo. La solución a este problema, la tienes en lo que te he escrito y en realizar un análisis adecuado.
